lib/elixir/lib/enum.ex

```diff
@@ -2247,18 +2247,21 @@ defmodule Enum do
       'fhjni'
 
   """
-  @spec take_random(t, integer) :: list
+  @spec take_random(t, non_neg_integer) :: list
   def take_random(_enumerable, 0), do: []
 
-  def take_random(first..last, 1) when first > last do
+  def take_random(first..first, 1) when is_integer(first),
+    do: [first]
+
+  def take_random(first..last, 1) when is_integer(first) and is_integer(last) and first > last do
     take_random(last..first, 1)
   end
 
-  def take_random(first..last, 1) do
+  def take_random(first..last, 1) when is_integer(first) and is_integer(last) do
```

Comment:
I would say that the `%Range{}` API is not public API? I mean it's visible in the documentation for `Range` (as `Range.t`) but still.